src/event-stream/event-stream.service.ts

```diff
@@ -131,16 +137,33 @@ export class EventStreamService {
   private readonly logger = new Logger(EventStreamService.name);
 
   private baseUrl: string;
+  private username: string;
+  private password: string;
 
   constructor(private http: HttpService) {}
 
-  configure(baseUrl: string) {
+  configure(baseUrl: string, username: string, password: string) {
     this.baseUrl = baseUrl;
+    this.username = username;
+    this.password = password;
+  }
+
+  private basicAuth() {
```

Comment:
Was planning on making this a utility function since it's used in `tokens` and `eventstreams`, but didn't see a project wide `utils`. Let me know what you think.

Reply:
Yea I think a top-level `utils` would be useful. Just hadn't needed one up to this point.